### DISCUSSION TITLE
Previews / wiki: provide .NET Cli build with patched DLLs

### DISCUSSION THREAD

**Discussion by dasMulli** [dotnet/csharplang#1868] [Q&A]
For getting more people to use the preview, do provide .NET Core CLI builds (or at least a zip + instructions for how to patch one).
E.g. this could be `2.1.402-nullable-ref-054`. To produce such a build, certain CLI build properties would have to be overwritten using MSBuild properties (at least this is what ProdCon does for producing builds with production references, maybe this could be used to set up additional "fork builds"?).
Consuming such a build would be done using a `global.json` file with a pinned reference.

**Comment by svick** [comment]
Unless something changed, the .Net Core CLI does support replacing the compiler with a different version, see https://github.com/dotnet/roslyn/issues/18783#issuecomment-308516907. So you don't really need a CLI build for that. Or do you think that's still too hard?

The issue is that I can't find a build of [the `Microsoft.Net.Compilers` package on Roslyn's MyGet](https://dotnet.myget.org/feed/roslyn/package/nuget/Microsoft.Net.Compilers) for the `features/NullableReferenceTypes` branch. Though as far as I can tell, that might change soon: https://github.com/ashmind/SharpLab/issues/343#issuecomment-420820066.

**Comment by dasMulli** [comment]
Yes per-project basis would work as well, but I was thinking more like at least having a zip of the files you'd need to replace in the SDK's `Roslyn` subdirectory from which msbuild and compiler assets are consumed.
Copying the SDK to a custom folder and replacing the file makes it available via global.json and you get all the expected build-server functionality (IIRC that's not the case with the compiler nuget package which always calls csc.dll/exe? or did that change?).
A custom build of the CLI would then also report its own version correctly but I don't really need that, it may just be good for feedback items to recognise the build by a known version number from `dotnet --info` output.
